### PR TITLE
[improve test]reboot SONiC if sanity check fail before run tset case

### PR DESCRIPTION
**Pull request by maggiemsft**
### Description of PR

reboot SONiC DUT if sanity check or interface check fail before running test case
also move decap test parameter to within test playbook, not doing it in test_sonic_by_testname.yml

### Type of change
- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Approach
- How did you do it?
Before: When trying to run test cases continuously, one test case failure due to crashed process or brought down interfaces will cause all subsequence tests fail.  
Now: If sanity check fail or interface check fail, reboot the SONiC DUT and try to bring back known good state. 
Thinking of doing restart process, since the goal is to re-establish a known good state, reboot is more clean. 

- How did you verify/test it?
verified it in my local testing environment

Any platform specific information?
Supported testbed topology if it's a new test case?

### CODE REVIEW

File: ansible/roles/test/tasks/base_sanity.yml [hunk 1]
```diff
@@ -1,20 +1,42 @@
...
+     - name: Get process information in syncd docker
+       shell: docker exec -i syncd ps aux | grep /usr/bin/syncd
+       register: psnew_out
+
+     - debug: var=psnew_out.stdout_lines
+
+     - name: Verify that syncd process is running
+       assert: { that: "{{ psnew_out.stdout_lines | length }} > 0"}
+
+     - name: Get oragent process information
```
Comment (jleveque) [inline ansible/roles/test/tasks/base_sanity.yml]: Typo: "oragent" --> "orchagent"

File: ansible/roles/test/tasks/base_sanity.yml [hunk 2]
```diff
@@ -1,20 +1,42 @@
...
+       shell: pgrep orchagent -a
+       register: orchnew_out
+
+     - debug: var=orchnew_out.stdout_lines
+
+     - name: Verify that orch process is running
```
Comment (jleveque) [inline ansible/roles/test/tasks/base_sanity.yml]: Might as well use the entire process name here ("orch" --> "orchagent").

File: ansible/roles/test/tasks/common_tasks/reboot_sonic.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+- name: reboot
+  become: true
+  shell: shutdown -r now "Reboot SONiC."
```
Comment (jleveque) [inline ansible/roles/test/tasks/common_tasks/reboot_sonic.yml]: Might as well make message more descriptive, e.g.:

`shutdown -r now "Warning! System is being rebooted remotely by reboot_sonic.yml."`

File: ansible/roles/test/tasks/common_tasks/reboot_sonic.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+- name: reboot
```
Comment (yxieca) [inline ansible/roles/test/tasks/common_tasks/reboot_sonic.yml]: Please rename this file to force_reboot.yml.

**Pull request merged**